Adding a built in Returns factor to the pipeline API

TimShawver opened the pull request:
I followed the RSI example in writing the test.  @ssanderson this is ready for you to look at when you get a chance.

ssanderson commented on tests/pipeline/test_factor.py:
```diff
@@ -222,3 +222,29 @@ def test_rsi(self, seed_value, expected):
+        # seed(seed_value)
+        # data = abs(randn(window_length, 3))
+        # expected = (data[-1] - data[0]) / data[0]
+        (100, array([-0.99581969, 0.78866002, 0.12724000]), 15),
```
I baked in the expected results for RSI because talib is an optional zipline dependency, so I couldn't just call it directly in the test.  For this, we should just compute the expected value directly in the test.

ssanderson commented on zipline/pipeline/factors/technical.py:
```diff
@@ -25,6 +25,22 @@
+
+    **Default Inputs**: [USEquityPricing.close]
+
+    **Default Window Length**: 30
```
I don't think this should have a default window_length.

ssanderson commented on zipline/pipeline/factors/technical.py:
```diff
@@ -25,6 +25,22 @@
+class Returns(CustomFactor):
+    """
+    Calculates the % change in close price over the given window_length
+    (i.e. the returns for a given asset).
```
I'd remove the parenthetical note here.

ssanderson commented on zipline/pipeline/factors/technical.py:
```diff
@@ -25,6 +25,22 @@
 
+class Returns(CustomFactor):
+    """
+    Calculates the % change in close price over the given window_length
```
`%` should probably be written out as `percent`.

TimShawver commented:
@ssanderson thanks for the feedback, I've made those change.  Let me know if there's anything else.

ssanderson commented:
LGTM

ssanderson commented:
@TimShawver can you add a whatsnew for this?

TimShawver commented:
@ssanderson I added a whatsnew and also added our built-in factors to the API docs for zipline.io.  I noticed they were missing when I tried to link to the new Returns factor from the release notes.
